src/main/kotlin/controllers/MainController.kt

```diff
@@ -225,4 +243,84 @@ class MainController constructor(
         worldRendererController.renderer.zLevelsSelected[z] = isSelected
         worldRendererController.renderer.isSceneUploadRequired = true
     }
+
+    private fun checkForUpdates() {
+        val now = System.currentTimeMillis() / 1000L
+        val lastChecked = configuration.getProp(SettingsController.LAST_CHECKED_FOR_UPDATES_PROP).toLongOrNull()
+
+        // see if it's been an hour since the last check
+        if (lastChecked != null && (now - lastChecked) < 3600) {
+            logger.info("Checked for updates within the past hour. Skipping check...")
+            return
+        }
+
+        try {
+            val url = URL("https://api.github.com/repos/ConnorDY/OSRS-Environment-Exporter/releases")
+
+            // create request
+            val conn = url.openConnection() as HttpsURLConnection
+            conn.requestMethod = "GET"
+            conn.setRequestProperty("Accept", "application/vnd.github.manifold-preview+json")
+            conn.useCaches = false
+            conn.doOutput = true
+
+            // capture the response
+            val inputStream = conn.inputStream
+            val bufferedReader = BufferedReader(InputStreamReader(inputStream))
+            val buffer = StringBuilder()
+
+            var line: String?
+            while (true) {
+                line = bufferedReader.readLine()
+                if (line != null) buffer.append(line)
+                else break
+            }
+            val data = buffer.toString()
+
+            // map the response to an array of GitHubRelease
+            val objectMapper = ObjectMapper()
+            val releases = objectMapper.readValue<Array<GitHubRelease>>(
+                data,
+                objectMapper.typeFactory.constructArrayType(GitHubRelease::class.java)
+            )
+
+            // determine if there is a newer version available
+            val currVersion = PackageMetadata.VERSION.split(".").map { it.toInt() }
+            var newerVersionUrl: String? = null
+
+            for (release in releases) {
+                val version = release.tagName.split(".").map { it.toInt() }
+
+                if (isVersionNewer(version, currVersion)) {
+                    newerVersionUrl = release.htmlURL
+                    break
+                }
+            }
+
+            if (newerVersionUrl != null) {
+                jMenuBar.add(
+                    JLinkLabel(
+                        newerVersionUrl,
+                        "Update available! Click here to download."
+                    )
+                )
+                jMenuBar.add(Box.createHorizontalStrut(4))
+                pack()
+            }
+        } catch (e: IOException) {
+            e.printStackTrace()
+        }
+
+        configuration.saveProp(SettingsController.LAST_CHECKED_FOR_UPDATES_PROP, now.toString())
+    }
+
+    private fun isVersionNewer(verA: List<Int>, verB: List<Int>): Boolean {
+        for ((ver1, ver2) in verA zip verB) {
```

Comment:
zip without padding will bias towards shorter version numbers. incidentally, I don't think kotlin provides zip with a padding feature. someone's made one though: https://stackoverflow.com/a/69050710/1091693 .

on a related note, there is a more efficient form of zip which is zip-with-transform, which just uses the functional style instead of a for-loop:
```kt
verA.zip(verB) { ver1, ver2 -> ... }
```